File: 2020/day3/part_one.go
package main

import (
	"bufio"
	"log"
	"os"
)

func main() {

	file, err := os.Open("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	input := make([]string, 0)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		input = append(input, scanner.Text())
	}

	cnt := 0

	i := 0
	rowLength := len(input[0])
	for row := range input {
		if row == 0 {
			continue
		}

		i += 3
		col := i % rowLength
		if string(input[row][col]) == "#" {
			cnt++
		}
	}

	log.Print(cnt)
}

File: 2021/day01/part_one.py
def main():
    cnt = 0
    previous_depth = None

    with open("input.txt") as f:
        for line in f:
            current_depth = int(line)
            if previous_depth is not None and current_depth > previous_depth:
                cnt += 1

            previous_depth = current_depth
            
    print(cnt)


if __name__ == "__main__":
    main()
